nats/pkg/server/nats/nats.go

```diff
@@ -0,0 +1,65 @@
+package nats
+
+import (
+	"context"
+	"time"
+
+	natsServer "github.com/nats-io/nats-server/v2/server"
+	stanServer "github.com/nats-io/nats-streaming-server/server"
+)
+
+type NATSServer struct {
+	ctx context.Context
+
+	natsOpts *natsServer.Options
+	stanOpts *stanServer.Options
+
+	server *stanServer.StanServer
+}
+
+func NewNATSServer(ctx context.Context, opts ...Option) (*NATSServer, error) {
+
+	server := &NATSServer{
+		ctx:      ctx,
+		natsOpts: &stanServer.DefaultNatsServerOptions,
+		stanOpts: stanServer.GetDefaultOptions(),
+	}
+
+	for _, o := range opts {
+		o(server.natsOpts, server.stanOpts)
+	}
+
+	return server, nil
+}
+
+func (n *NATSServer) ListenAndServe() (err error) {
+	n.server, err = stanServer.RunServerWithOpts(
+		n.stanOpts,
+		n.natsOpts,
+	)
+	if err != nil {
+		return err
+	}
+
+	defer n.Shutdown()
+
+	for {
+		// check if NATs server has an encountered an error
+		if err := n.server.LastError(); err != nil {
+			return err
+		}
+		// check if the NATs server is still running
+		if n.server.State() == stanServer.Shutdown {
+			return nil
+		}
+		// check if context was cancelled
+		if n.ctx.Err() != nil {
+			return nil
+		}
+		time.Sleep(1 * time.Second)
```

Comment:
Should we make this configurable? Or move it to a global var so we can at least overwrite on build?